test/e2e/e2e_test.go

```diff
@@ -863,3 +863,101 @@ var _ = Describe("e2e multiple external gateway update validation", func() {
 		}
 	})
 })
+
+func checkPodSleep(f *framework.Framework, nodeName, podName string, errChan chan error) {
+	defer GinkgoRecover()
+
+	contName := fmt.Sprintf("%s-container", podName)
+
+	command := []string{
+		"bash", "-c",
+		"sleep 10",
+	}
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    contName,
+					Image:   framework.AgnHostImage,
+					Command: command,
+				},
+			},
+			NodeName:      nodeName,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	podClient := f.ClientSet.CoreV1().Pods(f.Namespace.Name)
+	_, err := podClient.Create(pod)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	err = e2epod.WaitForPodNotPending(f.ClientSet, f.Namespace.Name, podName)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	result, err := podClient.Get(podName, metav1.GetOptions{})
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	framework.Logf("Created pod %q with IP %q %q", result.GetName(), result.Status.PodIP, result.Status.PodIPs)
+
+	err = e2epod.WaitForPodSuccessInNamespace(f.ClientSet, podName, f.Namespace.Name)
+
+	if err != nil {
+		logs, logErr := e2epod.GetPodLogs(f.ClientSet, f.Namespace.Name, pod.Name, contName)
+		if logErr != nil {
+			framework.Logf("Warning: Failed to get logs from pod %q: %v", pod.Name, logErr)
+		} else {
+			framework.Logf("pod %s/%s logs:\n%s", f.Namespace.Name, pod.Name, logs)
+		}
+	}
+
+	err = podClient.Delete(podName, &metav1.DeleteOptions{})
+
+	errChan <- err
+}
+
+var _ = Describe("e2e longevity test", func() {
+	var svcname = "nettest"
+
+	f := framework.NewDefaultFramework(svcname)
+
+	ginkgo.It("should create pods for a while without failing", func() {
+		ginkgo.By("Continuously creating pods for a while")
+
+		pod := 0
+		startTime := time.Now()
+
+		for time.Since(startTime) < 20*time.Minute {
+			// spin up 10 containers that each do nothing for 30 seconds
+
+			var errChans [10]chan error
+			for i, _ := range errChans {
+				errChans[i] = make(chan error)
+				go checkPodSleep(f, "", fmt.Sprintf("test-container-%v", pod), errChans[i])
+				pod++
+
+				time.Sleep(200 * time.Millisecond)
+			}
+
+			for _, v := range errChans {
+				select {
+				case err := <-v:
+					framework.ExpectNoError(err)
+				case <-time.After(300 * time.Second):
+					framework.Failf("Error waiting for pods")
```

Comment:
This will timeout serially for potential 300 seconds * 10 errChans because it's in the for loop. You can either use a single timer to check on all instances of the for loop iterations or you can do a time.AfterFunc before this for loop with a framework.Fail in it.

Reply:
it only will timeout is the channel does not return any value, I mean, if one of the channels blocks for more than 300 seconds the test will fail, it can only fail once, no?

Reply:
The test could drag on for 299 seconds * 10. Let me add a counter to ensure it times out after 5 total minutes of waiting.

Reply:
ahh ok, got it, sorry, 🙃